services/BasicTypes.proto

```diff
@@ -340,24 +340,49 @@ message CurrentAndNextFeeSchedule {
     FeeSchedule nextFeeSchedule = 2; // Contains next Fee Schedule
 }
 
+/* This contains the IP Address and the port representing an endpoint of a Node in a network */
+message NodeEndpoint {
+    string ipAddress = 1; // The IP address of the node
+    string port = 2; // The port of the node
+}
+
 /*
 The metadata for a Node – including IP Address, and the crypto account associated with the Node.
+
+Note :  This structure is intended to be used to parse 0.0.102, use NodeAddressForClients to parse 0.0.101.
+        Usage of NodeAddress to parse 0.0.101 is deprecated.
 */
 message NodeAddress {
-    bytes ipAddress = 1; // The ip address of the Node with separator & octets
-    int32 portno = 2; // The port number of the grpc server for the node
-    bytes memo = 3; // The memo field of the node (usage to store account ID is deprecated)
+    bytes ipAddress = 1 [deprecated=true]; // The ip address of the Node with separator & octets
+    int32 portno = 2 [deprecated=true]; // The port number of the grpc server for the node
+    bytes memo = 3 [deprecated=true]; // The memo field of the node (usage to store account ID is deprecated)
     string RSA_PubKey = 4; // The RSA public key of the node
     int64 nodeId = 5; // A non-sequential identifier for the node
     AccountID nodeAccountId = 6; // The account to be paid for queries and transactions sent to this node
     bytes nodeCertHash = 7; // A hash of the X509 cert used for gRPC traffic to this node
+    repeated NodeEndpoint nodeEndpoint = 8; // A node's IP address and port
+    string description = 9; // A description of the node. Max 100 bytes.
+    int64 stake = 10; // The amount of tinybars staked to this node
 }
 
 /*
 A list of nodes and their metadata.
 */
-message NodeAddressBook {
-    repeated NodeAddress nodeAddress = 1; // Contains multiple Node Address for the network
+message AddressBook {
+    repeated NodeAddress nodeAddress = 1; // Contains all details of the nodes for the network
+}
+
+/* All the information required to connect to the network.*/
+message NodeAddressForClients {
+    int64 nodeId = 5; // A non-sequential identifier for the node. This value is the key between the full and abbreviated address books
```

Comment:
Can you elaborate on this more please; reading the comments I'm not entirely sure what `nodeId` is; what are these `key`'s between full and abbreviated address books? Also, what is an abbreviated address book? I understand this field is a `int64` and `nodeAccountId` is an `AccountID`, but what is the difference between these two fields.

Reply:
The Idea behind this change is that, as we have multiple IPs for each node, when we get the addressBook 0.0.101 of mainnet for example.. we get a list with repeated entries for each node. To resolve this we want to create a new abbreviated version of the addressBook which will have a repeated `Endpoint`. So in future, when we get the addressbook we will get just enough information to connect to the network using this abbreviated message. 

Note: this  abbreviated message is `AddressBookForClients`.

A nodeId is just a sequential identifier of a node and the nodeAccountID is the `AccountID` associated with the node. Right now although they represent different things, their value is the same as in if nodeID is `N`, its account id is `0.0.N`

Reply:
> if nodeID is N, its account id is 0.0.N

Correction: if nodeID is N, its account id is 0.0.(N + 2). Example: node 1 is 0.0.3

Reply:
This is just the case now, but won't always be the case. Nodes can change their account ID in the future to something else while the node ID has to stay the same. Neither will necessarily stay sequential in the future either.